### ISSUE TITLE
Enum as query argument not properly managed

### ISSUE THREAD

**Issue by chrisvoo** [mghoneimy/php-graphql-oqm#9]
Hi,
let's assume we have the following schema:

```gql
  enum Provider {
    GOOGLE_PLACES
    WALLET_SAVER
  }

  autocompleteAddress(
    query: String!
    provider: Provider!
    istatCode: String
    limit: Int = 10
  ): [AddressDetails]
```
Launching  `./vendor/bin/generate_schema_objects` without arguments, I'll get the class `RootAutocompleteAddressArgumentsObject` with the following setter method `public function setProvider(ProviderInputObject $providerInputObject)` but that class is missing among the generated files.
So I had to create such file as the following (not sure it's the best way though):

```php
<?php

namespace SosTariffe\Components\GqlGenerator\SchemaObject;

use GraphQL\RawObject;
use GraphQL\SchemaObject\InputObject;

class ProviderInputObject extends InputObject
{
    private $value;

    public function __construct(string $enum)
    {
        $this->value = $enum;
    }

    public function __toString()
    {
        return $this->value;
    }

    public function toRawObject(): RawObject
    {
        return new RawObject($this->value);
    }
}
```

The final test produces a valid GraphQL query:

```php
        $rootObject = new RootQueryObject();

        $args = new RootAutocompleteAddressArgumentsObject();
        $args
            ->setIstatCode("50008")
            ->setLimit(5)
            ->setProvider(new ProviderInputObject(ProviderEnumObject::WALLET_SAVER))
            ->setQuery("Via Tosc");

        $addrDetails = $rootObject
            ->selectAutocompleteAddress($args);

        $addrDetails
            ->selectAddress()
            ->selectToponym()
            ->selectTown()
                ->selectName();

        $addrDetails->selectProviders()
                ->selectWalletSaver();

        echo $rootObject->getQuery();
```
And the result is the following:

```gql
{
  autocompleteAddress(query: "Via Tosc" provider: WALLET_SAVER istatCode: "50008" limit: 5) {
    address {
      toponym
      town {
        name
      }
    }
    providers {
      walletSaver
    }
  }
}
```

**Comment by mghoneimy** [comment]
Hello @chrisvoo , I'll look into this tonight, I remember fixing an issue similar to this recently.

**Comment by mghoneimy** [comment]
@chrisvoo Her's a link to this issue, it was reported before https://github.com/mghoneimy/php-graphql-oqm/issues/5.
It was fixed on v1.0.2, maybe I ask you which version are you using?

And to answer your question, the generator is not supposed to add the `ProviderInputObject` class specification to the `setProvider` method. It should leave it without type spec, that should be the method signature:
```public function setProvider($providerInputObject) {...```
You should be able to provide the value to the setter from the `ProviderEnumObject` class, which should have been created by the generator.
